packages/api/src/WithChangeListener.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { IChangeEvent } from './WithServerEvents';
+
+export interface IWithChangeListenerProps {
+  listenerKey: string;
+  filter: (change: IChangeEvent) => boolean;
+  read: () => void;
+  registerChangeListener: (
+    key: string,
+    listener: (event: IChangeEvent) => void
+  ) => void;
+  unregisterChangeListener: (key: string) => void;
+  children(props: {}): any;
+}
+
+export class WithChangeListener extends React.Component<
+  IWithChangeListenerProps
+> {
+  public constructor(props: IWithChangeListenerProps) {
+    super(props);
+    this.changeListener = this.changeListener.bind(this);
+  }
+
+  public render() {
+    return this.props.children(this.state);
```

Comment:
`this.state` will always be undefined here, no need to pass it.

Reply:
Derp, of course, thanks!  :-)